react/react.d.ts

```diff
@@ -41,7 +41,11 @@ declare module React {
         propTypes?: ValidationMap<P>;
     }
 
-    interface ComponentClass<P> extends ComponentStatics<P> {
+    interface ComponentClassType<P> extends ComponentStatics<P> {
+        new (): ComponentClass<P>;
+    }
+
+    interface ComponentClass<P> {
         // Deprecated in 0.12. See http://fb.me/react-legacyfactory
         // new(props: P): ReactElement<P>;
```

Comment:
OK, so perhaps what you really wanted to do was to uncomment this line? Could you revert this PR and do that instead?